Xamarin.Forms.Platform.Android/Renderers/CircularProgress.cs

```diff
@@ -0,0 +1,51 @@
+using System;
+using Android.Content;
+using Android.Util;
+using Android.Views;
+using AProgressBar = Android.Widget.ProgressBar;
+
+namespace Xamarin.Forms.Platform.Android
+{
+	internal class CircularProgress : AProgressBar, ViewTreeObserver.IOnPreDrawListener
+	{
+		public CircularProgress(Context context, IAttributeSet attrs, int defStyleAttr) : base(context, attrs, defStyleAttr)
+		{
+			ViewTreeObserver.AddOnPreDrawListener(this);
+			Indeterminate = true;
+		}
+
+		protected override void Dispose(bool disposing)
+		{
+			base.Dispose(disposing);
+			if (disposing && !this.IsDisposed())
+				ViewTreeObserver.RemoveOnPreDrawListener(this);
+		}
+
+		protected override void OnMeasure(int widthMeasureSpec, int heightMeasureSpec)
+		{
+			base.OnMeasure(widthMeasureSpec, heightMeasureSpec);
+
+			if (IsInEditMode)
```

Comment:
I have never used this before, but what does this do? If I think it does what it looks like, then this code only runs in the designer? Or is edit mode something different?

Reply:
Yes, it seems it was unnecessary. I've simplified this class, should work at least faster.